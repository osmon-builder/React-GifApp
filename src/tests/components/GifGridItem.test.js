import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from "../../Componentes/GifGridItem"

describe('GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)

    test('mostrar el componente correctamente',()=>
    {
    
      

      expect (wrapper).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim()).toBe(title);
    })

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        
        expect (img.prop('src')).toBe(url);
        expect (img.prop('alt')).toBe(title);
    })

    test('Debe de tener animate__flash', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');


        expect(className.includes('animate__flash')).toBe(true);
        
    })
    
    
    

})


