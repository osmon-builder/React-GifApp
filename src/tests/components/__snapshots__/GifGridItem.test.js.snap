// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__flash"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
